addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2587,6 +2588,62 @@ options.registry.ScrollButton = options.Class.extend({
     },
 });
 
+/**
+ * Allows for images to be replaced.
+ */
+options.registry.ReplaceImage = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        const $button = this.$el.find('we-button');
+        const $overlayArea = this.$overlay.find('.oe_snippet_remove');
+        $button.insertBefore($overlayArea);
+
+        return this._super(...arguments);
+    },
+    
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Replaces the image.
+     *
+     * @see this.selectClass for parameters
+     */
+    replaceImage: async function () {
+        // TODO: simulates a double click on an image from summernote,
+        // to be refactored when the new editor is merged
+        const $editable = this.$target.closest('.o_editable');
+        const options = $editable.closest('.o_editable, .note-editor').data('options');
+        const model = $editable.data('oe-model');
+        const field = $editable.data('oe-field');
+        const type = $editable.data('oe-type');
+        const dialog = new weWidgets.MediaDialog(this, 
+            {
+                res_model: model,
+                res_id: $editable.data('oe-id'),
+                domain: $editable.data('oe-media-domain'),
+                useMediaLibrary: field && (model === 'ir.ui.view' && field === 'arch' || type === 'html'),
+                onUpload: $editable.data('callbacks').onUpload,
+                noVideos: options && options.noVideos,
+            }, 
+            this.$target
+        );
+        return new Promise(resolve => {
+            dialog.on('save', this, (newMedia) => {
+                if (!newMedia) {
+                    return;
+                }
+                this.$target.replaceWith(newMedia);
+            });
+            dialog.on('closed', this, resolve);
+            dialog.open();
+        });
+    },
```

Comment:
Hum... why do you add a comment saying that this simulates a double click... if you don't simulate a double click ? :thinking: The point of my previous comment was to avoid duplicating code using summernote while waiting for the new editor by just double clicking on the image. So here, only `this.$target.dblclick()` should be enough for the whole method.

Reply:
Sorry for that ! It's changed. It made me realize that the double click does not work for the image gallery, the fix will be part of task-2472041

Reply:
I think @xO-Tx is already fixing that in one of its branches so don't worry about that yet :grimacing: